platform-manageability-agent/internal/comms/comms.go

```diff
@@ -0,0 +1,221 @@
+	}
+	cli.DMMgrClient = pb.NewDeviceManagementClient(cli.GrpcConn)
+	return nil
+}
+
+func ConnectToDMManager(ctx context.Context, serviceAddr string, tlsConfig *tls.Config) *Client {
+	dmMgr := NewClient(serviceAddr, tlsConfig)
+
+	cyclicalTicker := time.NewTicker(tickerInterval)
+	defer cyclicalTicker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Logger.Info("Connecting to DM Manager has been cancelled")
+			return nil
+		case <-cyclicalTicker.C:
+			err := dmMgr.Connect(ctx)
+			if err != nil {
+				log.Logger.Warnf("Can't connect to DM Manager, retrying: %v", err)
+				time.Sleep(dmMgr.RetryInterval)
+				continue
+			}
+			log.Logger.Info("Successfully connected to DM Manager")
+			return dmMgr
+		}
+	}
+}
+
+// parseAMTInfo parses the output of the `rpc amtinfo` command and populates the AMTStatusRequest.
+func parseAMTInfo(uuid string, output []byte) (*pb.AMTStatusRequest, error) {
+	var (
+		status  = pb.AMTStatus_DISABLED
+		version string
+	)
+
+	scanner := bufio.NewScanner(strings.NewReader(string(output)))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.HasPrefix(line, "Version") {
+			parts := strings.Split(line, ":")
+			if len(parts) > 1 {
+				version = strings.TrimSpace(parts[1])
+				status = pb.AMTStatus_ENABLED
+			}
+		}
+	}
+
+	req := &pb.AMTStatusRequest{
+		HostId:  strings.TrimSpace(string(uuid)),
+		Status:  status,
+		Version: version,
+	}
+	return req, nil
+}
+
+// ReportAMTStatus executes the `rpc amtinfo` command, parses the output, and sends the AMT status to the server.
+func (cli *Client) ReportAMTStatus(ctx context.Context) (pb.AMTStatus, error) {
+	defaultStatus := pb.AMTStatus_DISABLED
+
+	uuid, err := utils.GetSystemUUID()
+	if err != nil {
+		return defaultStatus, fmt.Errorf("failed to retrieve UUID: %w", err)
+	}
+
+	// TODO: RPC location need to be checked.
+	output, err := utils.ExecuteWithRetries("sudo", []string{"./rpc", "amtinfo"})
+	if err != nil {
+		req := &pb.AMTStatusRequest{
+			HostId:  uuid,
+			Status:  defaultStatus,
+			Version: "",
+		}
+		_, reportErr := cli.DMMgrClient.ReportAMTStatus(ctx, req)
+		if reportErr != nil {
+			return defaultStatus, fmt.Errorf("failed to report AMTStatus to DM Manager: %w", reportErr)
+		}
+		return defaultStatus, fmt.Errorf("failed to execute `rpc amtinfo` command: %w", err)
+	}
+
+	req, err := parseAMTInfo(uuid, output)
+	if err != nil {
+		return defaultStatus, fmt.Errorf("failed to parse `rpc amtinfo` output: %w", err)
+	}
+
+	_, err = cli.DMMgrClient.ReportAMTStatus(ctx, req)
+	if err != nil {
+		return defaultStatus, fmt.Errorf("failed to report AMT status: %w", err)
+	}
+	log.Logger.Info("Successfully reported AMT status")
+	return req.Status, nil
+}
+
+// RetrieveActivationDetails retrieves activation details and executes the activation command if required.
+func (cli *Client) RetrieveActivationDetails(ctx context.Context, conf *config.Config) error {
+	uuid, err := utils.GetSystemUUID()
```

Comment:
Each of these calls uses UUID, can we move the `utils.GetSystemUUID()` call to the main function then pass UUID as a parameter to each API call function? That way we only need to call it once